File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        Integer[] arr = Arrays.stream(citations).boxed().toArray(Integer[]::new);
        Arrays.sort(arr, (a, b) -> b - a);
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] < i + 1) return i;
        }
        return arr.length;
    }
}

File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    // 오름차순 정렬
    citations.sort((a, b) => a - b);

    let answer = 0;

    for (let i = 0; i < citations.length; i++) {
        const h = citations.length - i;

        if (citations[i] >= h) {
            answer = h;
            break;
        }
    }

    return answer;
}